src/Carbon/Carbon.php

```diff
@@ -1392,6 +1397,25 @@ public function toW3cString()
         return $this->format(static::W3C);
     }
 
+    /**
+     * Format the instance as RFC7231
+     *
+     * @return string
+     */
+    public function toRfc7231String()
+    {
+        if (!$this->utc) {
+            $tz = $this->getTimezone();
+            $this->setTimezone('GMT');
+        }
+        $formatted = $this->format(static::RFC7231);
+        if (isset($tz)) {
+            $this->setTimezone($tz);
+        }
+
+        return $formatted;
```

Comment:
Why not just `return $this->copy()->setTimezone('GMT')->format(static::RFC7231_FORMAT);` ?